Source/cryptalgo/SecureSocketPort.cpp

```diff
@@ -230,34 +262,44 @@ void SecureSocketPort::Handler::Update() {
 
         ASSERT(_ssl != nullptr);
 
+        // Client
         if (_handShaking == CONNECTING) {
             if ((result = SSL_connect(static_cast<SSL*>(_ssl))) == 1) {
-                _handShaking = EXCHANGE;
+                _handShaking = CONNECTED;
+                // If server has sent a certificate, and, we want to do 'our' own check
+                ValidateHandShake();
             }
-        }
+        } // Server
         else if (_handShaking == ACCEPTING) {
             if ((result = SSL_accept(static_cast<SSL*>(_ssl))) == 1) {
-                _handShaking = EXCHANGE;
+                _handShaking = CONNECTED;
```

Comment:
I think the ValidateHandshake will set the _handShaking correctly based on the outcome, so no real need to set it if you call the ValidateHandShake() method, it will become CONNECTED or ERROR I assume.

Reply:
ValidateHandshake returns void but overrides and thus resets the value internally to error. Should we transfer the responsibly of connected state to the method? What happens if we add additional and start using functionality (later) which does not fit the validation of certificates?